libpldmresponder/pdr_state_sensor.hpp

```diff
@@ -99,6 +101,65 @@ void generateStateSensorPDR(const DBusInterface& dBusIntf, const Json& json,
                 {
                     continue;
                 }
+                // now attach this entity to the container that was
+                // mentioned in the json, and add this entity to the
+                // parents entity assocation PDR
+
+                std::string parent_entity_path = e.value("parent_entity_path",
+                                                         "");
+                if (parent_entity_path != "" &&
+                    associatedEntityMap.contains(parent_entity_path))
+                {
+                    // find the parent node in the tree
+                    pldm_entity parent_entity =
+                        associatedEntityMap.at(parent_entity_path);
+                    pldm_entity child_entity = {pdr->entity_type,
+                                                pdr->entity_instance,
+                                                pdr->container_id};
+                    auto parent_node =
+                        pldm_entity_association_tree_find_with_locality(
+                            bmcEntityTree, &parent_entity, false);
+                    if (!parent_node)
+                    {
+                        // parent node not found in the entity association tree,
+                        // this should not be possible
+                        error(
+                            "Parent Entity of type {P_ENTITY_TYP} not found in the BMC Entity Association tree",
+                            "P_ENTITY_TYP",
+                            static_cast<unsigned>(parent_entity.entity_type));
+                        return;
+                    }
+                    pldm_entity_association_tree_add_entity(
+                        bmcEntityTree, &child_entity, pdr->entity_instance,
+                        parent_node, PLDM_ENTITY_ASSOCIAION_PHYSICAL, false,
+                        false, 0xFFFF);
+
+                    uint32_t bmc_record_handle = 0;
+                    auto lastLocalRecord = pldm_pdr_find_last_in_range(
+                        repo.getPdr(), BMC_PDR_START_RANGE, BMC_PDR_END_RANGE);
+                    bmc_record_handle = pldm_pdr_get_record_handle(
+                        repo.getPdr(), lastLocalRecord);
+                    [[maybe_unused]] uint8_t bmcEventDataOps;
```

Comment:
if we are not using this bmcEventDataOps, why define and assign values?

Reply:
We discussed earlier that we need to save the event ops changes, so we have saved it.